### PR TITLE
[Merged by Bors] - fix: do not count un-deprecations in `Mathlib/Deprecated`

### PR DESCRIPTION
**Pull request by adomani**
[Reported on Zulip](https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/Technical.20Debt.20Counters/near/482926966)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: scripts/technical-debt-metrics.sh [hunk 1]
```diff
@@ -79,7 +79,7 @@ titlesPathsAndRegexes=(
   "skipAssignedInstances flags"    "*"      "set_option tactic.skipAssignedInstances"
   "adaptation notes"               "*"      "adaptation_note"
   "disabled simpNF lints"          "*"      "nolint simpNF"
-  "disabled deprecation lints"     "*"      "set_option linter.deprecated false"
+  "disabled deprecation lints (outside of \`Deprecated\` dir)"     ":^Mathlib/Deprecated"      "set_option linter.deprecated false"
```
Comment (Vierkantor) [inline scripts/technical-debt-metrics.sh:82]: I think we don't need to mention this detail in the table itself, just like we don't mention that the maxHeartBeats modifications are only counted outside of the tests folder.
```suggestion
  "disabled deprecation lints"     ":^Mathlib/Deprecated"      "set_option linter.deprecated false"
```

Comment (Vierkantor) [reply scripts/technical-debt-metrics.sh:82]: I suppose we could add a footnote to the message posted by the bot if we want to be very clear about it.

**Review by Vierkantor — APPROVED** [review]
I left a suggestion but it looks good enough to me either way, thanks!

bors d+

**Comment by adomani** [comment]
Given how the discussion evolved in the Zulip chat, I am going to extract this using a variation of
```bash
echo "$(( "$(git grep -c "set_option linter.deprecated false" | awk -F: 'BEGIN{total=0} {total+=$2} END{print total}')" - "$(git grep -A1 "set_option linter.deprecated false" | grep -c "deprecated .*(since")"))"
```
that only searches outside the `Deprecated` dir.

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**